### DISCUSSION TITLE
[Feature request] Automatically prepend all paths in `navbar` and `sidebar` with root path of locale

### DISCUSSION THREAD

**Discussion by lifehackerhansol** [vuepress/ecosystem#128] [Q&A] [answered]
### Clear and concise description of the problem

As per #110, all paths in the theme configuration must be hardcoded. This makes it slightly difficult to manage several translated locales as all of them must be manually handled by the end user.

### Suggested solution

Since the [i18n page on VuePress](https://v2.vuepress.vuejs.org/guide/i18n.html) and [default theme locale documentation](https://ecosystem.vuejs.press/themes/default/config.html#locales) both suggest that a static path has to be passed anyway, I feel that it would make sense if the configurations themselves do not need to have the locale path hardcoded for `navbar` and `sidebar`, and be automatically prepended, reducing complexity on the user side.

For instance, if a user defines:
```js
    sidebar: {
        '/get-started.html': [
            ...
```
then it can be interpreted as `/zh_CN/get-started.html`.

### Alternative

At the moment, it is also impossible to use a variable for items in `sidebar` for some reason, i.e. this isn't possible and throws some error:
```js
const locale = 'zh_CN';
sidebar: {
    `/${locale}/get-started.html`: [
        ...
```

It would be nice if this was supported, as it would be easier to enable languages by just changing `$locale` somewhere.

**Comment by Mister-Hope** [comment]
CHATGPT or google is a good thing.

To use variable on property, you need a square bracket around it.

```js
const key = "class"
const value = "dark"

const attr = {
  [key]: value
}
```

Also, `zh_CN` is not in any standard format, while a standard one is `zh-CN`

**Comment by Mister-Hope** [comment]
Obviously,  I have no responsibility to teach you JavaScript basics here. So closing this one.

**Comment by Mister-Hope** [comment, marked as answer]
By the way,  If you write the grammar well, your alternative is our recommended approach about reusing navbar and sidebar across the locates. You can create a generator function for your navbar and sidebar , and reuse it.

E.g.: 

```ts
const getNavbar = (locales: string) => [
  `${locales}`,
  `${locales}guide/`,
  `${locales}about`,
]
```
